src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineFlexibleStaging.java

```diff
@@ -60,16 +66,33 @@ public void stageFlexible(StageFlexibleConfiguration config) throws AppEngineExc
           + config.getStagingDirectory().toPath());
     }
 
+    // verification for app.yaml that contains runtime:java
+    Path appYaml = config.getAppEngineDirectory().toPath().resolve("app.yaml");
+    String runtime = "non-determined";
+    try {
+      if (Files.isRegularFile(appYaml)) {
+        runtime = new AppYaml(appYaml).getRuntime();
+      }
+    } catch (IOException e) {
+      log.warning("Unable to determine runtime: error parsing app.yaml");
+    }
+
     try {
 
-      // Copy docker context to staging
       if (config.getDockerDirectory() != null && config.getDockerDirectory().exists()) {
-        if (!Files.isRegularFile(config.getDockerDirectory().toPath().resolve("Dockerfile"))) {
-          throw new AppEngineException("Docker directory " + config.getDockerDirectory().toPath()
-              + " does not contain Dockerfile");
+        if (runtime.equals("java")) {
+          log.warning("WARNING: Runtime 'java' detected, any docker configuration in "
```

Comment:
I would be more explicit and say:
```
WARNING: 'runtime: java' rather than 'runtime: custom' was detected in app.yaml. Any docker configuration in {} will be ignored.
```